src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -93,5 +94,63 @@ public virtual bool CanParseValue(object value)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Checks that the provided value is a valid default value.
+        /// This method should not throw an exception.
+        /// </summary>
+        /// <param name="value">The value to examine. Must not be <see langword="null"/>.</param>
+        public virtual bool IsValidDefault(object value)
+        {
+            try
+            {
+                return ToAST(value) != null;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Converts a value to an AST representation. This is necessary for introspection queries
+        /// to return the default values of this scalar type when used on input fields or field and directive arguments. This method may throw an exception
+        /// or return <see langword="null"/> for a failed conversion.
+        /// </summary>
+        /// <param name="value">The value to convert. Must not be <see langword="null"/>.</param>
+        /// <returns>AST representation of the specified value. Returning <see langword="null"/> indicates a failed conversion.</returns>
+        public virtual IValue ToAST(object value)
+        {
+            if (value == null)
+                return null;
+
+            try
+            {
+                var serialized = Serialize(value);
+                return serialized switch
+                {
+                    bool b => new BooleanValue(b),
+                    byte b => new IntValue(b),
+                    sbyte sb => new IntValue(sb),
+                    short s => new IntValue(s),
+                    ushort us => new IntValue(us),
+                    int i => new IntValue(i),
+                    uint ui => new LongValue(ui),
+                    long l => new LongValue(l),
+                    ulong ul => new BigIntValue(ul),
+                    BigInteger bi => new BigIntValue(bi),
+                    decimal d => new DecimalValue(d),
+                    float f => new FloatValue(f),
+                    double d => new FloatValue(d),
+                    string s => new StringValue(s),
+                    null => null,
+                    _ => throw new System.NotImplementedException($"Please override the '{nameof(ToAST)}' method of '{Name}' to support this operation.")
```

Comment:
```suggestion
                    _ => throw new System.NotImplementedException($"Please override the '{nameof(ToAST)}' method of '{GetType().Name}' scalar to support this operation.")
```

Reply:
The error description does not match any of our other descriptions.  They all use the name of the graph, not the name of the .NET class.  Why would you have this different?

Reply:
Because `override` and `method` words describe .NET class.

Reply:
Done